From 6da21a5924bb90b08a8642badf8898d087d70d5e Mon Sep 17 00:00:00 2001
From: NelloKudo <marshnelloosu@gmail.com>
Date: Wed, 12 Mar 2025 18:27:53 +0100
Subject: [PATCH] winex11: don't resize fullscreen windows

When using custom resolutions (ex. 1760x1080) in a 1920x1080 window, after alt-tab
the game's resolution resets to the window's on certain DEs/WMs. Returning before that happens is enough to fix it.

---
 dlls/winex11.drv/window.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dlls/winex11.drv/window.c b/dlls/winex11.drv/window.c
index 615fb9da086..6b1a21df390 100644
--- a/dlls/winex11.drv/window.c
+++ b/dlls/winex11.drv/window.c
@@ -1286,6 +1286,7 @@ static void window_set_config( struct x11drv_win_data *data, const RECT *new_rec
     data->desired_state.rect = *new_rect;
     if (!data->whole_window) return; /* no window, nothing to update */
     if (EqualRect( old_rect, new_rect ) && !above) return; /* rects are the same, no need to be raised, nothing to update */
+    if (data->is_fullscreen) return; /* keep the fullscreen window the same size */
 
     if (data->pending_state.wm_state == NormalState && data->net_wm_state_serial &&
         !(data->pending_state.net_wm_state & fullscreen_mask) &&
-- 
2.48.1

